examples/echoserver/echoserver.c

```diff
@@ -321,7 +320,29 @@ THREAD_RETURN CYASSL_THREAD echoserver_test(void* args)
         write_ssl = ssl;
 #endif
 
-        while ( (echoSz = CyaSSL_read(ssl, command, sizeof(command)-1)) > 0) {
+        while (1) {
+            do {
```

Comment:
Yes if a read/write (connect/accept) returns WC_PENDING_E then we need to poll. I think adding a blocking version of these API's is a good idea. I'd like to separate this cleanup and put it into a new PR. Not sure on the naming of new API's for this. Maybe "wolfSSL_readblock" or "wolfSSL_readasync" or "wolfSSL_read_ex"? There are many places in the examples we can refactor use this new API to cleanup the code.